PRCS/prc-5.md

```diff
@@ -0,0 +1,377 @@
+interface IUri {
+    /**
+     * @dev Returns the URI for token type `id`.
+     *
+     * If the `\{id\}` substring is present in the URI, it must be replaced by
+     * clients with the actual token type ID.
+     */
+    function uri(uint256 id) external view returns (string memory);
+}
+
+/// @dev A standard ERC1155 that can be burned and has a special uri function accepting a custom base URI.
+interface IInverseProjected1155 is IERC1155MetadataURI, IERC4906Agnostic {
+    /// @dev Emitted when `baseExtension` is updated from `oldBaseExtension` to `newBaseExtension`.
+    event SetBaseExtension(string oldBaseExtension, string newBaseExtension);
+
+    /// @dev Emitted when `baseUri` is updated from `oldUri` to `newUri`.
+    event SetBaseURI(string oldUri, string newUri);
+
+    /// @dev Burns `value` amount of token of ID `id` from transaction sender.
+    /// Reverts if transaction sender's balance of `id` is less than `value`.
+    function burn(uint256 id, uint256 value) external;
+
+    /// @dev Burns batch of `values` amounts of tokens of IDs `ids` from transaction sender.
+    /// Reverts if transaction sender's balance of any `id` is less than `value`.
+    function burnBatch(uint256[] memory ids, uint256[] memory values) external;
+
+    /// @dev Sets `_URI` as the `baseURI`.
+    /// Callable only by the contract owner.
+    /// Emits the `SetBaseURI` event.
+    function setBaseURI(string memory _URI) external;
+
+    /// @dev Sets `_newBaseExtension` as the `baseExtension`.
+    /// Callable only by the contract owner.
+    function setBaseExtension(string memory _newBaseExtension) external;
+
+    /// @dev Returns the token URI of specified `id` using a custom base URI.
+    function uri(uint256 id, string memory customBaseUri) external view returns (string memory);
+
+    /// @dev Returns the token URI of specified `id` using a call to contract implementing `IUri`.
+    function uri(uint256 id, IUri customUriInterface) external view returns (string memory);
+}
+
+/// @title Agnostic Metadata Update Extension
+interface IERC4906Agnostic {
+    /// @dev This event emits when the metadata of a token is changed.
+    /// So that the third-party platforms such as NFT market could
+    /// timely update the images and related attributes of the token.
+    event MetadataUpdate(uint256 _tokenId);
+
+    /// @dev This event emits when the metadata of a range of tokens is changed.
+    /// So that the third-party platforms such as NFT market could
+    /// timely update the images and related attributes of the tokens.
+    event BatchMetadataUpdate(uint256 _fromTokenId, uint256 _toTokenId);
+}
+```
+
+With the following baseURI:
+
+```bash
+https://${rpcBase}/inverseProjection/${standard}/${purpose}/
+```
+
+Where
+- `rpcBase` is the URI for the RPC
+- `standard` is for the specific PRC used to define the the format of this endpoint (ex: `prc5`)
+- `purpose` is a app-dependent string to describe what the NFT is for (ex: `gold`)
```

Comment:
we used `tgold` and not `gold` in Tarochi. Doesn't necessarily mean we need to change it here, but it might be better to avoid potential confusion